Wired up Network (ActivityPub) setting in Admin

sagzy opened the pull request:
ref https://linear.app/ghost/issue/PROD-2172

The Network toggle is disabled if:
1. (Ghost (Pro) only) the feature is disabled by config
2. The site is hosted on a subdirectory, localhost or an IP address in production

sagzy committed:
Wired up Network (ActivityPub) setting in Admin

ref https://linear.app/ghost/issue/PROD-2172

The Network toggle is disabled if:
1. (Ghost (Pro) only) the feature is disabled by config
2. The site is hosted on a subdirectory, localhost or an IP address in production

ErisDS commented on apps/admin-x-settings/src/components/settings/growth/Network.tsx:
```diff
@@ -1,22 +1,69 @@
+    const [isDisabledByConfig, setIsDisabledByConfig] = useState(false);
+    useEffect(() => {
+        async function checkLimiter() {
+            if (limiter?.isLimited('limitSocialWeb') && await limiter.checkWouldGoOverLimit('limitSocialWeb')) {
```
Curious why we chose to use `checkWouldGoOverLimit()` here instead of `isDisabled()`?

ErisDS reviewed (commented):
Left one question, but otherwise LGTM!

sagzy committed:
Used sync .isDisabled() over async .checkWouldGoOverLimit()

- the async method was used to prevent a race condition: toggle would be checked (but disabled) even if the feature was limited by config
- adding the `key` prop forces React to unmount the Toggle component whenever the key value changes, and it now renders the correct checked/disabled states
